Enabling ZC143 Display Support on the MIMXRT700_EVK

EmilioCBen opened the pull request:
Enabling ZC143 Display Support on the MIMXRT700_EVK

JarmouniA commented on dts/bindings/display/chipone,co5300.yaml:
```diff
@@ -0,0 +1,34 @@
+# Copyright 2025 NXP
+# SPDX-License-Identifier: Apache-2.0
+
```
```suggestion

title: CO5300 display controller

```

JarmouniA commented on dts/bindings/display/chipone,co5300.yaml:
```diff
@@ -0,0 +1,34 @@
+    type: phandle-array
+    description: |
+      The BLn pin is asserted to control the backlight of the panel.
+      The sensor receives this as an active-high signal.
```
This binding is for the controller, not the panel, bl gpio should be controlled by the MCU's display controller driver, or in the `board.c` file.
See an example 
https://github.com/zephyrproject-rtos/zephyr/blob/b677e828d0750a25a6923a0ec594a9fb06585101/boards/shields/st_lcd_dsi_mb1835/st_lcd_dsi_mb1835.overlay#L93
https://github.com/zephyrproject-rtos/zephyr/blob/b677e828d0750a25a6923a0ec594a9fb06585101/drivers/display/display_stm32_ltdc.c#L284

JarmouniA replied:
Yes many other panel controllers are doing it, but it is wrong and we will have to correct them.

EmilioCBen replied:
In that case, I am not sure if this PR is the anchor point to start doing architecture changes.
I would not not want to update the mipi driver and cause potential failures to shields already using it.
I only need to support this display using currently supplied architecture of zephyr. If you feel this correction needs
to occur can this change not happen on a separate PR that does a mass clean up for the backlight gpio?

EmilioCBen commented on drivers/display/Kconfig.co5300:
```diff
@@ -0,0 +1,10 @@
+	select MIPI_DSI
+	depends on DT_HAS_CHIPONE_CO5300_ENABLED
+	help
+	  Enable driver for CO5300 display
```
I have added controller to this line.

JarmouniA commented on drivers/display/display_co5300.c:
```diff
@@ -0,0 +1,474 @@
+		return gpio_pin_set_dt(&config->backlight_gpios, 1);
+	} else {
+		return -ENOTSUP;
+	}
```
```suggestion
	}
	
	return -ENOTSUP;
```

JarmouniA commented on dts/bindings/display/chipone,co5300.yaml:
replace "sensor" with "controller".

JarmouniA reviewed (dismissed):
Add an entry for the controller in https://github.com/zephyrproject-rtos/zephyr/blob/540278c1e824a6fa93e222cbbf536d35fa57b4a1/tests/drivers/build_all/display/app.overlay#L398
And for the shield in https://github.com/zephyrproject-rtos/zephyr/blob/540278c1e824a6fa93e222cbbf536d35fa57b4a1/tests/drivers/display/display_check/testcase.yaml#L58

EmilioCBen committed:
tests: drivers: Adding zc143ac72mipi to display tests

Updated display tests to run the zc143ac72mipi shield.

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>

EmilioCBen committed:
drivers: display: Add co5300 support.

Added driver controller co5300 in an effort
to support the zc143ac72mipi shield.

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>

EmilioCBen committed:
boards: shield: Add shield support for zc143ac72mipi

Adding shield support for zc143ac72mipi.
A bridge board for the co5300 display controller.

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>

decsny commented:
@JarmouniA as far as the backlight GPIO. IMO it makes even less sense to put it on the MCU peripheral node. the backlight GPIO is for the display. We don't have nodes for the display panel in DT right now. The "display controller" node basically represents "a display driven by this display controller". The backlight control is also in the display API, not the mipi or whatever else API. So it makes most architectural sense to put it where it is now IMO.

decsny commented:
twister failure fixed by https://github.com/zephyrproject-rtos/zephyr/pull/98153
